third_party/xla_client/pjrt_computation_client.h

```diff
@@ -174,6 +179,41 @@ class PjRtComputationClient : public ComputationClient {
     std::shared_ptr<PjRtBuffer> buffer;
   };
 
+  struct PjRtReplicatedData : public Data {
```

Comment:
I'm confused. Why would this wrapper help given it's just barely a vector of PjRtData?

Reply:
In the existing implementation, we only track the DataPtr for a single device's copy of the replicated data across steps, so after each step we discard the DataPtr for the replicated data on all other devices. This makes us `CopyToDevice` to all other devices every step. Additionally, CopyToDevice will block if the devices are busy (i.e. executing the computation), which causes the additional overhead.

PjRtReplicatedData just provides a Data class that tracks all of the replicated handles, but the benefit is that we can track all devices' DataPtr across steps. This lets us only CopyToDevice once in the `TransferReplicatedDataToServer` call and update the replicated data values by `Assign` after the computation when the output is replicated.